OptimizelySDK/Odp/LruCache.cs

```diff
@@ -0,0 +1,171 @@
+            TimeSpan? itemTimeout = null, ILogger logger = null
+        )
+        {
+            const int DEFAULT_MAX_SIZE = 10000;
+            const int CACHE_DISABLED = 0;
+            var defaultTimeout = TimeSpan.FromMinutes(10);
+
+            _mutex = new object();
+
+            if (maxSize is null)
+            {
+                _maxSize = DEFAULT_MAX_SIZE;
+            }
+            else if (maxSize < 0)
+            {
+                _maxSize = CACHE_DISABLED;
+            }
+            else
+            {
+                _maxSize = maxSize.Value;
+            }
+
+            if (itemTimeout is null)
+            {
+                _timeout = defaultTimeout;
+            }
+            else if (itemTimeout?.TotalMilliseconds < 0)
+            {
+                _timeout = _timeoutDisabled;
+            }
+            else
+            {
+                _timeout = itemTimeout.Value;
+            }
+
+            _logger = logger ?? new DefaultLogger();
+
+            _cache =
+                new Dictionary<string, (LinkedListNode<string> node, ItemWrapper value)>(_maxSize);
+            _list = new LinkedList<string>();
+        }
+
+        public void Save(string key, T value)
+        {
+            if (_maxSize == 0)
+            {
+                return;
+            }
+
+            lock (_mutex)
+            {
+                if (_cache.ContainsKey(key))
+                {
+                    (LinkedListNode<string> node, ItemWrapper item) = _cache[key];
+                    _list.Remove(node);
+                    _list.AddFirst(node);
+                    _cache[key] = (node, item);
+                }
+                else
+                {
+                    if (_cache.Count >= _maxSize)
+                    {
+                        var removeKey = _list.Last.Value;
+                        _cache.Remove(removeKey);
+                        _list.RemoveLast();
+                    }
+
+                    _cache.Add(key, (_list.AddFirst(key), new ItemWrapper(value)));
+                }
+            }
+        }
+
+        public T Lookup(string key)
+        {
+            if (_maxSize == 0)
+            {
+                return default;
+            }
+
+            lock (_mutex)
+            {
+                if (!_cache.ContainsKey(key))
+                {
+                    return default;
+                }
+
+                (LinkedListNode<string> node, ItemWrapper item) = _cache[key];
+
+                var currentTimestamp = DateTime.Now.MillisecondsSince1970();
+
+                if (_timeout == _timeoutDisabled ||
+                    (currentTimestamp - item.CreationTimestamp < _timeout.TotalMilliseconds))
+                {
+                    _list.Remove(node);
+                    _list.AddFirst(node);
+
+                    _cache[key] = (node, item);
+
+                    return item.Value;
+                }
+
+                _cache.Remove(key);
```

Comment:
should it be in `else` statement?
If this is not caught by unit test, please add a unit test for it.

Reply:
I think the `else` is redundant.  The item was expired and needed to be removed from both the index (`LinkedList`) and the cache itself. 

The unit test `ShouldHandleWhenItemsExpire` should be covering this case.